builder/simplify.cc

```diff
@@ -1341,26 +1341,38 @@ class simplifier : public node_mutator {
   }
 
   void visit(const block* op) override {
-    stmt a = mutate(op->a);
-    stmt b = mutate(op->b);
-
-    const if_then_else* a_if = a.as<if_then_else>();
-    const if_then_else* b_if = b.as<if_then_else>();
+    std::vector<stmt> stmts;
+    stmts.reserve(op->stmts.size());
+    bool changed = false;
+    for (const stmt& s : op->stmts) {
+      stmts.push_back(mutate(s));
+      changed = changed || !stmts.back().same_as(s);
+    }
+
+    // If any consecutive entries are if-then-elses with the same conditions,
```

Comment:
I think this logic is hard to verify as correct, and we now don't actually exercise this code path except for one test that doesn't check the new edge cases. I think we should just get rid of it until we find we need it. I think maybe we should even just get rid of `if_then_else` completely, we currently don't use it at all.